tests/AndroidXMigrationTests/Tests/ManagedMigrationTests.cs

```diff
@@ -143,5 +144,28 @@ public void EmbeddedJavaArtifactsAreMigrated(string supportDll, string layoutFil
 				}
 			}
 		}
+
+		[Theory]
+		[InlineData(ActiveDirectoryDll, ActiveDirectoryPdb, CecilMigrationResult.ContainedSupport | CecilMigrationResult.PotentialJavaArtifacts)]
+		[InlineData(ManagedSupportDll, ActiveDirectoryPdb, CecilMigrationResult.ContainedSupport | CecilMigrationResult.PotentialJavaArtifacts | CecilMigrationResult.ContainedJavaArtifacts)]
+		[InlineData(ManagedSupportDll, ManagedSupportDll, CecilMigrationResult.ContainedSupport | CecilMigrationResult.PotentialJavaArtifacts | CecilMigrationResult.ContainedJavaArtifacts)]
+		public void CanRunMigrationOnCorruptedPdbs(string assembly, string symbols, CecilMigrationResult expectedResult)
+		{
+			var dll = Utils.GetTempFilename(Path.GetFileName(assembly));
+			var pdb = Path.ChangeExtension(dll, "pdb");
+			File.Copy(assembly, dll, true);
+			File.Copy(symbols, pdb, true);
+			
+			var migratedDll = RunMigration(dll, expectedResult);
+
+			using (var def = AssemblyDefinition.ReadAssembly(migratedDll))
+			{
+				var types = def.MainModule.GetTypeReferences();
+				Assert.Empty(types.Where(t => 
+					(t.FullName.StartsWith("Android.Support.") || t.FullName.StartsWith("Android.Arch.")) &&
```

Comment:
In case someone adds Material. Do we need to add it in the condition?

Reply:
Yeah. Not the best of tests...

Let me extract this logic into a function that can be reused. I'll keep it manual for more control for now.